src/test/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/nonassociations/NonAssociationsResourceIntTest.kt

```diff
@@ -0,0 +1,807 @@
+        offenderAtLeeds.nomsId,
+      )
+        .headers(setAuthorisation(roles = listOf("ROLE_NOMIS_NON_ASSOCIATIONS")))
+        .body(BodyInserters.fromValue(updateNonAssociationRequest().copy(type = "INVALID")))
+        .exchange()
+        .expectStatus().isBadRequest
+        .expectBody().jsonPath("$.userMessage").value<String> {
+          assertThat(it).contains("Type with code=INVALID does not exist")
+        }
+    }
+
+    @Test
+    fun `effectiveDate in the future`() {
+      callCreateEndpoint()
+      webTestClient.put().uri(
+        "/non-associations/offender/{offenderNo}/ns-offender/{nsOffenderNo}",
+        offenderAtMoorlands.nomsId,
+        offenderAtLeeds.nomsId,
+      )
+        .contentType(MediaType.APPLICATION_JSON)
+        .headers(setAuthorisation(roles = listOf("ROLE_NOMIS_NON_ASSOCIATIONS")))
+        .body(BodyInserters.fromValue(updateNonAssociationRequest().copy(effectiveDate = LocalDate.now().plusDays(1))))
+        .exchange()
+        .expectStatus().isBadRequest
+        .expectBody().jsonPath("$.userMessage").value<String> {
+          assertThat(it).contains("Effective date must not be in the future")
+        }
+    }
+
+    @Test
+    fun `comment too long`() {
+      callCreateEndpoint()
+      webTestClient.put().uri(
+        "/non-associations/offender/{offenderNo}/ns-offender/{nsOffenderNo}",
+        offenderAtMoorlands.nomsId,
+        offenderAtLeeds.nomsId,
+      )
+        .contentType(MediaType.APPLICATION_JSON)
+        .headers(setAuthorisation(roles = listOf("ROLE_NOMIS_NON_ASSOCIATIONS")))
+        .body(BodyInserters.fromValue(updateNonAssociationRequest().copy(comment = "x".repeat(241))))
+        .exchange()
+        .expectStatus().isBadRequest
+        .expectBody().jsonPath("$.userMessage").value<String> {
+          assertThat(it).contains("Comment is too long (max allowed 240 characters)")
+        }
+    }
+
+    @Test
+    fun `invalid date should return bad request`() {
+      val invalidSchedule = validUpdateJsonRequest().replace(
+        """"effectiveDate" : "2023-02-28"""",
+        """"effectiveDate" : "2023-13-27"""",
+      )
+      callCreateEndpoint()
+      webTestClient.put().uri(
+        "/non-associations/offender/{offenderNo}/ns-offender/{nsOffenderNo}",
+        offenderAtMoorlands.nomsId,
+        offenderAtLeeds.nomsId,
+      )
+        .headers(setAuthorisation(roles = listOf("ROLE_NOMIS_NON_ASSOCIATIONS")))
+        .contentType(MediaType.APPLICATION_JSON)
+        .body(BodyInserters.fromValue(invalidSchedule))
+        .exchange()
+        .expectStatus().isBadRequest
+        .expectBody().jsonPath("$.userMessage").value<String> {
+          assertThat(it).contains("Invalid value for MonthOfYear (valid values 1 - 12): 13")
+        }
+    }
+
+    @Test
+    fun `will update non-association with correct details`() {
+      callCreateEndpoint()
+      callUpdateEndpoint(offenderAtMoorlands.nomsId, offenderAtLeeds.nomsId)
```

Comment:
same here - hard to tell what callUpdateEndpoint(offenderAtMoorlands.nomsId, offenderAtLeeds.nomsId) does